src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -563,6 +563,22 @@ public function has($id)
      */
     public function get($id, $invalidBehavior = ContainerInterface::EXCEPTION_ON_INVALID_REFERENCE)
     {
+        if ($this->isCompiled()) {
+            $id = $this->normalizeId($id);
+
+            if (isset($this->definitions[$id]) && $this->definitions[$id]->isPrivate()) {
+                @trigger_error(sprintf('Fetching the "%s" private service is deprecated and will fail in Symfony 4.0. Make the service public instead.', $id), E_USER_DEPRECATED);
+            }
+            if (isset($this->aliasDefinitions[$id]) && $this->aliasDefinitions[$id]->isPrivate()) {
+                @trigger_error(sprintf('Fetching the "%s" private alias is deprecated and will fail in Symfony 4.0. Make the alias public instead.', $id), E_USER_DEPRECATED);
```

Comment:
these deprecation are required because in master, these services/aliases are removed by the `RemoveUnusedServicesPass`.